Databases/VSSTester/VSSTester.ps1

```diff
@@ -40,167 +88,125 @@ param(
-            Invoke-EnableVSSTracing
-            Invoke-CreateExTRATracingConfig
-            Invoke-EnableExTRATracing
-            Invoke-DiskShadow #---
-            Get-VSSWritersAfter
-            Invoke-RemoveExposedDrives #---
-            Invoke-DisableExTRATracing
-            Invoke-DisableDiagnosticsLogging
-            Invoke-DisableVSSTracing
-            Get-WindowsEventLogs
-        } elseif ($Selection -eq 2) {
-            Get-ExchangeVersion
-            Get-VSSWritersBefore
-            Get-Databases
-            Get-DBtoBackup
-            Get-CopyStatus
-            Invoke-EnableDiagnosticsLogging
-            Invoke-EnableVSSTracing
-            Invoke-CreateExTRATracingConfig
-            Invoke-EnableExTRATracing
-
-            #Here is where we wait for the end user to perform the backup using the backup software and then come back to the script to press "Enter", thereby stopping data collection
-            Get-Date
-            Write-Host "Data Collection" -ForegroundColor green $nl
-            Write-Host "--------------------------------------------------------------------------------------------------------------"
-            " "
-            Write-Host "Data collection is now enabled." -ForegroundColor Yellow
-            Write-Host "Please start your backup using the third party software so the script can record the diagnostic data." -ForegroundColor Yellow
-            Write-Host "When the backup is COMPLETE, use the <Enter> key to terminate data collection..." -ForegroundColor Yellow -NoNewline
-            Read-Host
-
-            Invoke-DisableExTRATracing
-            Invoke-DisableDiagnosticsLogging
-            Invoke-DisableVSSTracing
-            Get-VSSWritersAfter
-            Get-WindowsEventLogs
+
+            # Only collect event logs if we exited the loop gracefully
+            $collectEventLogs = $true
         }
     } finally {
-        # always stop our transcript at end of script's execution
-        # we catch a failure here if we try to stop a transcript that's not running
-        try {
-            " " + $nl
-            Get-Date
-            Write-Host "Stopping transcript log..." -ForegroundColor Green $nl
-            Write-Host "--------------------------------------------------------------------------------------------------------------"
-            " "
-            Stop-Transcript
-            " " + $nl
-            do {
-                Write-Host
-                $continue = Read-Host "Please use the <Enter> key to exit..."
-            }
-            while ($null -notmatch $continue)
-            exit
-        } catch { }
+        Write-Host "$(Get-Date) Stopping traces..."
+        Invoke-DisableExTRATracing -ServerName $serverName -Database $dbForBackup -OutputPath $LoggingPath
+        Invoke-DisableDiagnosticsLogging
+        Invoke-DisableVSSTracing
+        Write-Host "$(Get-Date) Tracing stopped."
+
+        if ($collectEventLogs) {
```

Comment:
Maybe we should add in an `else` statement with a `Write-Host` stating that we aren't collecting data because `$collectEventLogs` isn't set.

Reply:
Done.